### PR TITLE
Make CookieTempDataProvider the default ITempDataProvider

### LINKED ISSUE #5766: Make CookieTempDataProvider the default TempData provider
All in the title.

cc @DamianEdwards @danroth27

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #5766.

Also added an extension for adding SessionsStateTempDataProvider.

### CODE REVIEW

**Comment by rynowak** [comment]
This is definitely an announcement-worthy change

**Comment by ryanbrandenburg** [comment]
Announced in aspnet/Announcements#225.

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcBuilderExtensions.cs [hunk 1]
```diff
@@ -67,6 +67,28 @@ public static IMvcBuilder AddViewComponentsAsServices(this IMvcBuilder builder)
         }
 
         /// <summary>
+        /// Registers <see cref="SessionStateTempDataProvider"/> as the default <see cref="ITempDataProvider"/>
+        /// in the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcBuilder"/>.</param>
+        /// <returns>The <see cref="IMvcBuilder"/>.</returns>
+        public static IMvcBuilder AddSessionStateTempDataProvider(this IMvcBuilder builder)
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcBuilderExtensions.cs]: The functional tests's BasicWebSite's  Startup file need to be updated to reflect this default temp data provider change.

Comment (kichalla) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcBuilderExtensions.cs]: Update the BasicWebsite's startup file to reflect this default provider change

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/WebSites/BasicWebSite/StartupWithSessionTempDataProvider.cs [hunk 2]
```diff
@@ -6,13 +6,14 @@
 
 namespace BasicWebSite
 {
-    public class StartupWithCookieTempDataProvider
+    public class StartupWithSessionTempDataProvider
```
Comment (kichalla) [inline test/WebSites/BasicWebSite/StartupWithSessionTempDataProvider.cs]: Rename this file and class to `Startup` and the other to `StartupWithSessionTempDataProvider` as currently this class does not have the session services or middleware

File: test/WebSites/BasicWebSite/StartupWithSessionTempDataProvider.cs [hunk 3]
```diff
@@ -6,18 +6,21 @@
...
     {
         public void ConfigureServices(IServiceCollection services)
         {
+            // CookieTempDataProvider is the default ITempDataProvider, so we must override it with session
```
Comment (Eilon) [inline test/WebSites/BasicWebSite/StartupWithSessionTempDataProvider.cs]: Way too many trailing spaces!

**Pull request merged**